### PR TITLE
Updating poco and making submodule

### PR DESCRIPTION
**Pull request by proller**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "contrib/poco"]
+	path = contrib/poco
+	url = https://github.com/ClickHouse-Extras/poco
```
Comment (vavrusa) [inline .gitmodules]: Are you using 1.8-based Poco? There's a https://github.com/pocoproject/poco/issues/1487 with the older versions (fails to build on OS X). I patched just the file in https://github.com/vavrusa/ClickHouse/commit/7cef1d02e9c45494a802751c3ae8b0d44393e7ff but updating the whole library would be superior.

Comment (proller) [reply .gitmodules]: no, poco/develop branch with c++11 support

File: libs/libdaemon/src/BaseDaemon.cpp [hunk 2]
```diff
@@ -812,6 +812,12 @@ void BaseDaemon::initialize(Application & self)
     /// Ставим terminate_handler
     std::set_terminate(terminate_handler);
 
+    {
```
Comment (alexey-milovidov) [inline libs/libdaemon/src/BaseDaemon.cpp]: Missing comment.

**Comment by alexey-milovidov** [comment]
contrib/libpoco was not deleted.

**Comment by alexey-milovidov** [comment]
CMAKE_BUILD_TYPE was suddenly changed to Release.

```CMAKE_BUILD_TYPE:STRING=Release```

We have RelWithDebInfo as default.

**Comment by vavrusa** [comment]
The vendored version breaks build with the latest OS X clang:

```
clickhouse/libs/libcommon/src/JSON.cpp:459:52: error: call to 'format' is ambiguous
        throw JSONException("JSON: array index " + Poco::NumberFormatter::format(n) + " out of bounds.");
                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
clickhouse/contrib/poco/Foundation/include/Poco/NumberFormatter.h:236:21: note: candidate function
        static std::string format(bool value, BoolFormat format = FMT_TRUE_FALSE);
                           ^
clickhouse/contrib/poco/Foundation/include/Poco/NumberFormatter.h:413:37: note: candidate function
inline std::string NumberFormatter::format(int value)
                                    ^
clickhouse/contrib/poco/Foundation/include/Poco/NumberFormatter.h:453:37: note: candidate function
inline std::string NumberFormatter::format(unsigned value)
                                    ^
clickhouse/contrib/poco/Foundation/include/Poco/NumberFormatter.h:579:37: note: candidate function
inline std::string NumberFormatter::format(Int64 value)
                                    ^
clickhouse/contrib/poco/Foundation/include/Poco/NumberFormatter.h:619:37: note: candidate function
inline std::string NumberFormatter::format(UInt64 value)
                                    ^
clickhouse/contrib/poco/Foundation/include/Poco/NumberFormatter.h:659:37: note: candidate function
inline std::string NumberFormatter::format(float value)
                                    ^
clickhouse/contrib/poco/Foundation/include/Poco/NumberFormatter.h:683:37: note: candidate function
inline std::string NumberFormatter::format(double value)
```

**Comment by alexey-milovidov** [comment]
Hope this will help: https://github.com/yandex/ClickHouse/commit/10392d54523b8eb2f8892f52220f350fb1f61096

**Comment by vavrusa** [comment]
@alexey-milovidov It affects more places where `UInt64` and `size_t` were used interchangeably. Like:

```
In file included from /clickhouse/dbms/src/Parsers/IAST.cpp:4:
In file included from /clickhouse/dbms/src/IO/Operators.h:5:
/clickhouse/dbms/src/IO/ReadHelpers.h:112:5: error: no matching function for call to 'readVarUInt'
    readVarUInt(size, buf);
    ^~~~~~~~~~~
/clickhouse/dbms/src/IO/VarInt.h:78:13: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'UInt32 &' (aka 'unsigned int &') for 1st argument
inline void readVarUInt(UInt32 & x, ReadBuffer & istr)
            ^
/clickhouse/dbms/src/IO/VarInt.h:92:13: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'UInt16 &' (aka 'unsigned short &') for 1st argument
inline void readVarUInt(UInt16 & x, ReadBuffer & istr)
            ^
/clickhouse/dbms/src/IO/VarInt.h:112:13: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'UInt64 &' (aka 'unsigned long long &') for 1st argument
inline void readVarUInt(UInt64 & x, ReadBuffer & istr)
            ^
/clickhouse/dbms/src/IO/VarInt.h:130:13: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'UInt64 &' (aka 'unsigned long long &') for 1st argument
inline void readVarUInt(UInt64 & x, std::istream & istr)
            ^
/clickhouse/dbms/src/IO/VarInt.h:143:21: note: candidate function not viable: requires 3 arguments, but 2 were provided
inline const char * readVarUInt(UInt64 & x, const char * istr, size_t size)
```

Or conversions from `Field`, etc. Not sure if we should alias the type like it was before, or fix all those uses to use `Poco::UInt64`.

**Comment by vavrusa** [comment]
https://github.com/yandex/ClickHouse/pull/1356

**Pull request merged**